components/nimbus/ios/Nimbus/FeatureHolder.swift

```diff
@@ -218,7 +218,12 @@ public extension FMLFeatureInterface {
 
     func toJSONString() -> String {
         let encoder = JSONEncoder()
-        let data = try! encoder.encode(self)
-        return String(decoding: data, as: UTF8.self)
+        do {
+            let data = try encoder.encode(self)
+            return String(data: data, encoding: .utf8) ?? ""
+        } catch {
+            print("Error encoding to JSON: \(error)")
+            return ""
+        }
```

Comment:
Chatgpt suggests this:

```
func toJSONString() -> String {
    let encoder = JSONEncoder()
    let data = try! encoder.encode(self)
    return String(data: data, encoding: .utf8) ?? fatalError("Failed to convert data to String using UTF-8 encoding")
}
```

@adudenamedruby Does this look like it preserves the original behaviour in a way that would make the linter satisfied?

Reply:
> Eh, no, because this will cause the app to crash. As unfun as it may be, we should be implementing proper error handling.

I think crashing _is_ the correct behavior here, though:

* `FMLFeatureInterface` (`self`) is `Encodable`, with an automatic implementation that the Swift compiler synthesizes for us. If `encode()` fails for it, I think that indicates a serious logic error in the Swift compiler. `JSONEncoder.encode()` takes arbitrary values, so it can't be non-throwing in the general case; but in our case, we know that it's valid, or the compiler wouldn't have been able to synthesize an implementation for us.
* `JSONEncoder.encode()` must return valid UTF-8, because that is a requirement of the JSON data format. If `encode()` returns invalid UTF-8, that's also a serious bug in the Swift standard library.